cognite/client/experimental/model_hosting/models.py

```diff
@@ -0,0 +1,397 @@
+        self.name = item["name"]
+        self.error_msg = item["errorMsg"]
+        self.model_id = item["modelId"]
+        self.created_time = item["createdTime"]
+        self.metadata = item["metadata"]
+        self.source_package_id = item["sourcePackageId"]
+        self.status = item["status"]
+        self.description = item["description"]
+        self.project = item["project"]
+
+
+class ModelVersionCollectionResponse(CogniteCollectionResponse):
+    _RESPONSE_CLASS = ModelVersionResponse
+
+
+class ModelArtifactResponse(CogniteResponse):
+    def __init__(self, internal_representation):
+        super().__init__(internal_representation)
+        item = self.to_json()
+        self.name = item["name"]
+        self.size = item["size"]
+
+
+class ModelArtifactCollectionResponse(CogniteCollectionResponse):
+    _RESPONSE_CLASS = ModelArtifactResponse
+
+
+class ModelLogResponse(CogniteResponse):
+    def __init__(self, internal_representation):
+        super().__init__(internal_representation)
+        item = self.to_json()
+        self.prediction_logs = item["predict"]
+        self.training_logs = item["train"]
+
+    def to_json(self):
+        """Returns data as a json object"""
+        return self.internal_representation["data"]
+
+
+class PredictionError(APIError):
+    pass
+
+
+class ModelsClient(APIClient):
+    def __init__(self, **kwargs):
+        super().__init__(version="0.6", **kwargs)
+        self._LIMIT = 1000
+
+    def create_model(
+        self,
+        name: str,
+        description: str = "",
+        metadata: Dict[str, Any] = None,
+        input_fields: List[str] = None,
+        output_fields: List[str] = None,
+    ) -> ModelResponse:
+        """Creates a new model
+
+        Args:
+            name (str):             Name of model
+            description (str):      Description
+            metadata (Dict[str, Any]):          Metadata about model
+            input_fields (List[str]):   List of input fields the model accepts
+            output_fields (List[str]:   List of output fields the model produces
+
+        Returns:
+            Dict: The created model.
+        """
+        url = "/analytics/models"
+        model_body = {
+            "name": name,
+            "description": description,
+            "metadata": metadata or {},
+            "inputFields": input_fields or [],
+            "outputFields": output_fields or [],
+        }
+        res = self._post(url, body=model_body)
+        return ModelResponse(res.json())
+
+    def list_models(self, limit: int = None, cursor: int = None, autopaging: bool = False) -> ModelCollectionResponse:
+        """List all models.
+
+        Args:
+            limit (int): Maximum number of models to return. Defaults to 250.
+            cursor (str): Cursor to use to fetch next set of results.
+            autopaging (bool): Whether or not to automatically page through all results. Will disregard limit.
```

Comment:
This is more of a question for the future moving towards 1.0 and @f1cognite: Maybe pagination should be completely hidden for the user and auto pagination always on? Resulting in just a limit parameter specifying how long the auto pagination will go. And then if the user would like to stream, support generators as well or something.

Reply:
Agree, that would be neat.